packages/terra-alert/src/Alert.jsx

```diff
@@ -0,0 +1,280 @@
+  /**
+   * Child Nodes providing the message content for the alert. Can contain text and HTML.
+   */
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
+  /**
+   * The icon to be used for an alert of type custom.
+   */
+  customIcon: PropTypes.element,
+  /**
+   * The status bar color to be used for an alert of type custom.
+   */
+  customStatusColor: PropTypes.string,
+  /**
+   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause the Dismiss button to be included on the alert.
+   */
+  onDismiss: PropTypes.func,
+  /**
+   * The title for the alert which will be bolded.
+   */
+  title: PropTypes.string,
+  /**
+   * The type of alert to be rendered. One of Alert.Types.ALERT, Alert.Types.ERROR, Alert.Types.WARNING, Alert.Types.ADVISORY,
+   * Alert.Types.INFO, Alert.Types.SUCCESS, Alert.Types.GAP_CHECKING, Alert.Types.OUTSIDE_RECORDS, Alert.Types.CUSTOM.
+   * Use the Types attribute of the Alert component for access to these type strings.
+   */
+  type: PropTypes.oneOf([
+    AlertTypes.ALERT,
+    AlertTypes.ERROR,
+    AlertTypes.WARNING,
+    AlertTypes.ADVISORY,
+    AlertTypes.INFO,
+    AlertTypes.SUCCESS,
+    AlertTypes.GAP_CHECKING,
+    AlertTypes.OUTSIDE_RECORDS,
+    AlertTypes.CUSTOM,
+  ]),
+};
+
+const defaultProps = {
+  alertAction: null,
+  children: '',
+  customIcon: null,
+  customStatusColor: '',
+  onDismiss: null,
+  title: '',
+  type: AlertTypes.ALERT,
+};
+
+const getAlertClassName = (type) => {
+  switch (type) {
+    case AlertTypes.ALERT:
+      return styles.alertType;
+    case AlertTypes.ERROR:
+      return styles.errorType;
+    case AlertTypes.WARNING:
+      return styles.warningType;
+    case AlertTypes.ADVISORY:
+      return styles.advisoryType;
+    case AlertTypes.INFO:
+      return styles.infoType;
+    case AlertTypes.SUCCESS:
+      return styles.successType;
+    case AlertTypes.GAP_CHECKING:
+      return styles.gapCheckingType;
+    case AlertTypes.OUTSIDE_RECORDS:
+      return styles.outsideRecordsType;
+    case AlertTypes.CUSTOM:
+      return styles.customType;
+    default:
+      return '';
+  }
+};
+
+const getAlertIcon = (type, customIcon) => {
+  switch (type) {
+    case AlertTypes.ALERT:
+      return (<IconAlert height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.ERROR:
+      return (<IconError height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.WARNING:
+      return (<IconWarning height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.ADVISORY:
+      return null;
+    case AlertTypes.INFO:
+      return (<IconInformation height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.SUCCESS:
+      return (<IconSuccess height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.GAP_CHECKING:
+      return (<IconRequired height="1.3333rem" width="1.3333rem" />);
+    case AlertTypes.OUTSIDE_RECORDS:
+      return (<IconDiamond height="1.3333rem" width="1.3333rem" />);
```

Comment:
This was a late addition from @neilpfeiffer.  See the approved pattern here: https://wiki.ucern.com/pages/viewpage.action?spaceKey=UserExperience&title=Terra+UI%3A+Approved+Patterns